### PR TITLE
Fix implementation issue of convTranspose2d to support options.groups

### PR DESCRIPTION
**Pull request by BruceDai**
@fdwr @huningxin PTAL, thanks.

### CODE REVIEW

File: src/conv_transpose2d.js [hunk 1]
```diff
@@ -115,8 +116,8 @@ export function convTranspose2d(
             ih += realStrideHeight, ++oh) {
             for (let iw = -realBeginningPaddingWidth, ow = 0; ow < outputWidth;
               iw += realStrideWidth, ++ow) {
-              const effectiveOutputChannel = oc + g * outputChannelsPerGroup;
-              const outputLocation = [ib, effectiveOutputChannel, oh, ow];
```
Comment (huningxin) [inline src/conv_transpose2d.js:119]: Keep these two lines and use `oc` directly to access filter value at line 143?
```js
const filterValue = filter.getValueByLocation(
    [oc, ic, filterHeight - realKh - 1, filterWidth - realKw - 1]);
```

File: src/lib/validate-input.js [hunk 2]
```diff
@@ -99,6 +99,28 @@ export function validateConv2dParams(input, filter, {bias, groups = 1}) {
   }
 }
 
+export function validateConvTranspose2dParams(input, filter, {bias, groups = 1}) {
+  const inputChannels = input.shape[1];
+  // filter of oihw
+  const outputChannelsPerGroup = filter.shape[0];
+
```
Comment (huningxin) [inline src/lib/validate-input.js:106]: ```const outputChannels = outputChannelsPerGroup * groups```

File: src/lib/validate-input.js [hunk 3]
```diff
@@ -99,6 +99,28 @@ export function validateConv2dParams(input, filter, {bias, groups = 1}) {
...
+  if (filter.rank !== 4) {
+    throw new Error('The filter should be a 4-D tensor.');
+  }
+  if (inputChannels % groups !== 0) {
+    throw new Error('The input channels is invalid.');
+  }
+  if (inputChannels !== filter.shape[1]) {
+    throw new Error('The input channels of filter is invalid.');
+  }
+  if (bias && (bias.rank !== 1 || bias.shape[0] != outputChannelsPerGroup * groups)) {
```
Comment (huningxin) [inline src/lib/validate-input.js:119]: Use `outputChannels`

File: src/conv_transpose2d.js (hunk 1 above)
Comment (BruceDai) [reply src/conv_transpose2d.js]: Done.

**Comment by BruceDai** [comment]
Thanks @huningxin for reviewing. I've updated commit to address your comments, please take another look.

**Review by huningxin — APPROVED** [review]
LGTM, thanks!

**Review by fdwr — APPROVED** [review]

**Pull request merged**